server/src/test/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzerTests.java

```diff
@@ -0,0 +1,526 @@
+                        stats.addProximity(fieldLookup.getPostingsField(file), bytes);
+                        break;
+                    case "kdm":
+                        // not a per-field file, but we can hackishly do this for the points case.
+                        for (ObjectLongCursor<String> e : readPointLengths(directory, file, sis, reader.getFieldInfos())) {
+                            stats.addPoints(e.key, e.value);
+                        }
+                        break;
+                    case "fdt":
+                    case "fdx":
+                    case "fdm":
+                        // We don't have per field Codec for stored, vector, and norms field
+                        stats.addStoredField("_all_stored_fields", bytes);
+                        break;
+                    case "tvx":
+                    case "tvd":
+                        stats.addTermVectors("_all_vectors_fields", bytes);
+                        break;
+                    case "nvd":
+                    case "nvm":
+                        stats.addNorms("_all_norms_fields", bytes);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        } finally {
+            if (directory != reader.directory()) {
+                IOUtils.close(directory);
+            }
+        }
+    }
+
+    private static ObjectLongMap<String> readPointLengths(Directory dir, String fileName, SegmentInfo sis,
+                                                          FieldInfos fieldInfos) throws IOException {
+        assert IndexFileNames.getExtension(fileName).equals("kdm") : fileName;
+        final ObjectLongMap<String> pointLengths = new ObjectLongHashMap<>();
+        final long totalDataLength = dir.fileLength(fileName.substring(0, fileName.length() - 1) + "d");
+        try (ChecksumIndexInput in = dir.openChecksumInput(fileName, IOContext.READONCE)) {
+            // fail hard if its not exactly the version we do this hack for.
+            CodecUtil.checkIndexHeader(in, "Lucene86PointsFormatMeta", 0, 0, sis.getId(), "");
+            int lastFieldNum = -1;
+            long lastMinFP = 0;
+            while (true) {
+                int fieldNum = in.readInt();
+                if (fieldNum == -1) {
+                    break;
+                }
+                CodecUtil.checkHeader(in, BKDWriter.CODEC_NAME, BKDWriter.VERSION_CURRENT, BKDWriter.VERSION_CURRENT);
+                in.readVInt(); // total dims
+                int indexDims = in.readVInt(); // index dims
+                in.readVInt(); // max points in leaf node
+                int bytesPerDim = in.readVInt(); // bytes per dim
+                in.readVInt(); // num leaves
+                in.skipBytes(2L * indexDims * bytesPerDim); // min/max values
+                in.readVLong(); // point count
+                in.readVInt(); // num docs
+                int numIndexBytes = in.readVInt();
```

Comment:
Reading Lucene's file format feels a bit heroic to me and high-maintenance.

Reply:
Ok, I removed this and added a approximate test for points in https://github.com/elastic/elasticsearch/pull/74051/commits/4fd2cd11a9e2cfeb8432157deb325145d3d4fea1